aerogear/mobile-security-service-operator
# Update Deployment GVK; fix CRD group; tighten permissions

## Motivation
JIRA: https://issues.jboss.org/browse/AEROGEAR-9616

- The operator ServiceAccount shouldn't be given permissions that it doesn't need.
- We shouldn't use a domain that we don't control for the group.
- We should use the latest version of the `Deployment` kind.

## What
This change was originally just intended to tighten the permissions, but I also took the opportunity to update to use the new version for `Deployment` (it took me a while to realise why `Deployment` wasn't working originally even when I had it specified in the role), and update the group part for the CRDs (we don't own `aerogear.com`, so that was likely just a typo originally).

## Why
Wouldn't that already be covered by the earlier "Motivation" section?

## How
See the "Files changed" tab, or navigate to https://github.com/aerogear/mobile-security-service-operator/pull/155/files in your browser.

## Verification Steps
- Check out this branch
- `sed -i -e "s|quay.io/aerogear/mobile-security-service-operator:master|quay.io/aerogear/mobile-security-service-operator:$(git rev-parse --short HEAD)|" deploy/operator.yaml`
- `make install`
- Make sure the service provisions correctly, and there are no error logs related to permissions in either the operator or service pods.
- Provision an app CR and make sure that works too, and there are no errors.

## Checklist:
